src/Analyzer.Utilities/DiagnosticHelpers.cs

```diff
@@ -98,5 +99,117 @@ public static string GetMemberName(ISymbol symbol)
             // For other language constructs
             return symbol.Name;
         }
+               
+        public static bool IsInvisibleOutsideAssemblyAtSymbolLevel(ISymbol symbol)
+        {
+            return SymbolIsPrivateOrInternal(symbol)
+                || SymbolIsProtectedInSealed(symbol);
+        }
+
+        public static bool SymbolIsPrivateOrInternal(ISymbol symbol)
+        {
+            var access = symbol.DeclaredAccessibility;
+            return access == Accessibility.Private
+                || access == Accessibility.Internal
+                || access == Accessibility.ProtectedAndInternal
+                || access == Accessibility.NotApplicable;
+        }
+
+        public static bool SymbolIsProtectedInSealed(ISymbol symbol)
+        {
```

Comment:
I am surprised to find that these 3 functions are never used in Analyzers. I will remove it.